jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/UserDataLoader.java

```diff
@@ -0,0 +1,190 @@
+package com.jme3.scene.plugins.gltf;
+
+import com.jme3.plugins.json.JsonArray;
+import com.jme3.plugins.json.JsonElement;
+import com.jme3.plugins.json.JsonObject;
+import com.jme3.plugins.json.JsonPrimitive;
+import com.jme3.scene.Spatial;
+/*
+ * $Id$
+ *
+ * Copyright (c) 2019, Simsilica, LLC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import java.lang.reflect.Array;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Attaches GLTF "extras" data to objects as appropriate.
+ *
+ * @author Paul Speed
+ */
+public class UserDataLoader implements ExtrasLoader {
+
+  private static final Logger log = Logger.getLogger(
+    UserDataLoader.class.getName()
+  );
+
+  public UserDataLoader() {}
+
+  @Override
+  public Object handleExtras(
+    GltfLoader loader,
+    String parentName,
+    JsonElement parent,
+    JsonElement extras,
+    Object input
+  ) {
+    log.fine(
+      "handleExtras(" +
+      loader +
+      ", " +
+      parentName +
+      ", " +
+      parent +
+      ", " +
+      extras +
+      ", " +
+      input +
+      ")"
+    );
```

Comment:
Regarding formating, to me, I would like to place them in one line until the line length exceeds the max length, only then I move them to the new line. Just my own preference, not sure about others though!